src/router.js

```diff
@@ -139,6 +139,88 @@ router.put('/registrations/:id', async (request, response) => {
   }
 });
 
+router.get('/returns', async (request, response) => {
+  try {
+    const returns = await Return.findAll();
+    if (returns) {
+      // If they are, send back the finalised registrations.
+      response.status(200).send(returns);
+    } else {
+      response.status(404).send({message: `No returns found.`});
+      return;
+    }
+  } catch (error) {
+    response.status(500).send({error});
+  }
+});
+
+router.get('/registrations/:id/return', async (request, response) => {
+  try {
+    // Try to parse the incoming ID to make sure it's really a number.
+    const existingId = Number(request.params.id);
+    if (isNaN(existingId)) {
+      response.status(404).send({message: `Registration ${request.params.id} not valid.`});
+      return;
+    }
+
+    // Check if there's a registration allocated at the specified ID.
+    const existingReg = await Registration.findOne(existingId);
+    if (existingReg === undefined || existingReg === null) {
+      response.status(404).send({message: `Registration ${existingId} not allocated.`});
+      return;
+    }
+
+    const returns = await Return.findRegReturns(existingId);
+
+    if (returns) {
+      // If they are, send back the finalised registration.
+      response.status(200).send(returns);
+    } else {
+      response.status(404).send({message: `Registration ${request.params.id} not valid.`});
+      return;
+    }
```

Comment:
Not a big fan of the returning from both halves of an if-else, especially if it can be re-written to not need an else at all. You can do something like the pattern already used above.

```diff
- const returns = await Return.findRegReturns(existingId);
- 
- if (returns) {
-   // If they are, send back the finalised registration.
-   response.status(200).send(returns);
- } else {
-   response.status(404).send({message: `Registration ${request.params.id} not valid.`});
-   return;
- }
+ const returns = await Return.findRegReturns(existingId);
+ if (returns === undefined || returns === null) {
+   return response.status(404).send({message: `Registration ${request.params.id} not valid.`});
+ }
+ 
+ return response.status(200).send(returns);
```